bokeh/plotting/helpers.py

```diff
@@ -83,65 +88,66 @@ def _pop_colors_and_alpha(glyphclass, kwargs, prefix="", default_alpha=1.0):
+            continue
         result[argname] = kwargs.pop(prefix + argname, color)
 
     # NOTE: text fill color should really always default to black, hard coding
     # this here now until the stylesheet solution exists
     if "text_color" in glyphclass.properties():
         result["text_color"] = kwargs.pop(prefix + "text_color", "black")
 
-    alpha = kwargs.pop(prefix+"alpha", default_alpha)
+    alpha = kwargs.pop(prefix + "alpha", default_alpha)
     for argname in ("fill_alpha", "line_alpha", "text_alpha"):
-        if argname not in glyphclass.properties(): continue
+        if argname not in glyphclass.properties():
+            continue
         result[argname] = kwargs.pop(prefix + argname, alpha)
 
     return result
 
 
-def _process_legend_kwargs(kwargs):
-    if kwargs.get('legend') and kwargs.get('label'):
-        raise RuntimeError("Cannot set both legend and label at the same time.")
+def _get_legend_item_label(kwargs):
     legend = kwargs.pop('legend', None)
     source = kwargs.get('source')
+    legend_item_label = None
     if legend:
         if isinstance(legend, string_types):
-            # Try and do something intelligent with a legend string:
-            # * value if it's not in column data source
-            # * field if it is
+            # Do the simple thing first
+            legend_item_label = value(legend)
+            # But if there's a source - try and do something smart
             if source and hasattr(source, 'column_names'):
                 if legend in source.column_names:
-                    kwargs['label'] = legend
-                else:
-                    kwargs['label'] = value(legend)
+                    legend_item_label = field(legend)
         else:
-            # Otherwise just accept folks were being intentional
-            kwargs['label'] = legend
+            legend_item_label = legend
+    return legend_item_label
+
 
 _GLYPH_SOURCE_MSG = """
 Supplying a user-defined data source AND iterable values to glyph methods is deprecated.
 
 See https://github.com/bokeh/bokeh/issues/2056 for more information.
 """
 
+
 def _process_sequence_literals(glyphclass, kwargs, source, is_user_source):
     dataspecs = glyphclass.dataspecs_with_props()
     for var, val in kwargs.items():
 
         # ignore things that are not iterable
-        if not isinstance(val, Iterable): continue
-
+        if not isinstance(val, Iterable):
+            continue
         # pass dicts (i.e., values or fields) on as-is
-        if isinstance(val, dict): continue
-
+        if isinstance(val, dict):
+            continue
         # let any non-dataspecs do their own validation (e.g., line_dash properties)
-        if var not in dataspecs: continue
-
+        if var not in dataspecs:
+            continue
         # strings sequences are handled by the dataspec as-is
-        if isinstance(val, string_types): continue
-
+        if isinstance(val, string_types):
+            continue
```

Comment:
I really prefer the old version, I find the whitespace between each pair of lines much more readable